clang-tools-extra/clang-tidy/misc/UnusedUsingDeclsCheck.cpp

```diff
@@ -71,6 +72,10 @@ void UnusedUsingDeclsCheck::registerMatchers(MatchFinder *Finder) {
                          templateArgument().bind("used")))),
                      this);
   Finder->addMatcher(userDefinedLiteral().bind("used"), this);
+  Finder->addMatcher(
+      elaboratedType(unless(hasQualifier(nestedNameSpecifier())),
+                     hasUnqualifiedDesugaredType(type().bind("usedType"))),
```

Comment:
```suggestion
      loc(elaboratedType(unless(hasQualifier(nestedNameSpecifier())),
                     hasUnqualifiedDesugaredType(type().bind("usedType")))),
```

To avoid matching too much.